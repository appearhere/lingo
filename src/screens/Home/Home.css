.root {
  height: 100%;
  background-color: var(--color-greyLightest);
}

.iconContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 4rem;
  width: 100%;
  height: 100%;
  flex-direction: column;
}

.icon {
  font-size: inherit;
}
