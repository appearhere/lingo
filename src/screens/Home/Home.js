import React from 'react';
import PropTypes from 'prop-types';
import Fuse from 'fuse.js';
import qs from 'query-string';
import flow from 'lodash/fp/flow';
import sortBy from 'lodash/fp/sortBy';
import groupBy from 'lodash/fp/groupBy';
import map from 'lodash/map';

import TermList from '../../components/TermList/TermList';

import terms from '../../terms.json';

import sharedCss from '../../shared.css';

const fuse = new Fuse(terms, {
  shouldSort: true,
  threshold: 0.2,
  location: 0,
  distance: 100,
  maxPatternLength: 32,
  minMatchCharLength: 1,
  keys: ['name', 'definition'],
});

const formatResults = flow(sortBy('name'), groupBy('department'));

const Home = ({ match, location }) => {
  const { q: query } = qs.parse(location.search);
  const currentDepartment = match.params.department;
  const searchResults = query ? fuse.search(query) : terms;
  const results = formatResults(searchResults);

  return (
    <div className={ sharedCss.container }>
      { currentDepartment
        ? <TermList department={ currentDepartment } terms={ results[currentDepartment] } />
        : map(results, (groupedTerms, department) => (
          <TermList key={ department } department={ department } terms={ groupedTerms } />
      )) }
    </div>
  );
};


Home.propTypes = {
  match: PropTypes.shape({}),
  location: PropTypes.shape({}),
};

export default Home;
