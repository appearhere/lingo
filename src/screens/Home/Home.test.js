import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter, Route } from 'react-router';

import Home from './Home';

jest.mock('../../utils/terms', () => ([]));

describe('Home component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<StaticRouter><Route component={ Home } /></StaticRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
