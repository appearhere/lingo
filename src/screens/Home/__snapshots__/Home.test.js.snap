// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
<div
  className=""
>
  <h1
    className={undefined}
  >
     Here's the Lingo
  </h1>
  <p
    className={undefined}
  >
    Lingo is here to help you understand the terms we all use.
  </p>
  <section>
    <div
      className={undefined}
    />
  </section>
</div>
`;
